docker/pusher.go

```diff
@@ -90,6 +90,10 @@ func NewCredentialsProvider(
 		}
 	}
 
+	// Creating an instance of the fn.Client ensures that the config path
+	// exists:
+	_ = fn.New()
```

Comment:
This awkwardness is a result of the reliance on static functions previously for asserting something that is perhaps more appropriately an instance-scoped data.  In particular, the effective config path can vary (and will vary more in the future with the Global Config option `WithConfigPath`).  Therefore, these should be methods on an instance of a Client rather than package-static members of `fn`.  This will be improved in a separate PR.